nmstatectl/nmstatectl.py

```diff
@@ -90,12 +116,49 @@ def _filter_interfaces(state, patterns):
     return showinterfaces
 
 
-def show(args):
-    state = netinfo.show()
-    if args.only != '*':
-        patterns = [p for p in args.only.split(',')]
+def _filter_state(state, whitelist):
+    if whitelist != '*':
+        patterns = [p for p in whitelist.split(',')]
         state['interfaces'] = _filter_interfaces(state, patterns)
+    return state
+
+
+def edit(args):
+    state = _filter_state(netinfo.show(), args.only)
+
+    editor = os.environ.get('EDITOR', 'vi')
+    if args.yaml:
+        suffix = '.yaml'
+        txtstate = pretty_yaml(state)
+    else:
+        suffix = '.json'
+        txtstate = pretty_json(state)
+    with tempfile.NamedTemporaryFile(suffix=suffix,
+                                     prefix='nmstate-') as statefile:
+        statefile.write(txtstate)
+        statefile.flush()
+
+        try:
+            subprocess.check_call([editor, statefile.name])
 
+            statefile.seek(0)
+
+            if args.yaml:
+                new_state = yaml.load(statefile)
+            else:
+                new_state = json.load(statefile)
+
+            print('Applying the following state: ')
+            print_state(new_state, use_yaml=args.yaml)
+
+            netapplier.apply(new_state, verify_change=args.verify)
```

Comment:
Please handle exception from `netapplier.apply()` by print error message instead of traceback.

Reply:
What is wrong with a traceback? It is more informative.

Reply:
A traceback will trigger abrt and traceback text itself might also scare non-tech people.

Reply:
We need the traceback to determine the problem.
If you would like to build an error library that translates exceptions into nice messages and somehow provides the needed info for debugging, please add an issue for it. I think it has low priority at the moment.
Most python tools I know transparently leave the exceptions as is and do not try to wrap them for users. Perhaps it makes sense to wrap general errors we expect users to encounter.

Reply:
We could add a `--verbose` or `--debug` flag that would enable tracebacks and not show them otherwise. Dominik also noticed the traceback in the `--verify` demo and I agree that it does not look so nice.

Reply:
Since this should happen in a similar way for `nmstatectl set`, let's do this in a separate PR.